src/main/java/net/minestom/server/particle/ParticleCreator.java

```diff
@@ -23,14 +30,26 @@ public static ParticlePacket createParticlePacket(Particle particleType, boolean
         } else {
             data = new byte[0];
         }
+
         return new ParticlePacket(particleType.id(), distance, x, y, z, offsetX, offsetY, offsetZ, particleData, count, data);
     }
 
-    public static ParticlePacket createParticlePacket(Particle particleType,
+
+    public static ParticlePacket createParticlePacket(Particle particleType, boolean distance,
+                                                      double x, double y, double z,
+                                                      float offsetX, float offsetY, float offsetZ,
+                                                      float particleData, int count, ParticleOptions options) {
+
+        Check.argCondition(particleType != options.type(), "ParticleOptions mismatch!");
```

Comment:
I'd recommend this error message be a bit more clear, something lie `"Particle type " + particleType.namespace() + " does not support ParticleOptions " + options != null ? options.getClass().getName() : "no options";`

Also, you should probably have some `@Nullable` and `@NotNull` annotations on the method parameters.

Reply:
I improved the error message and enforced `@NotNull` arguments but I don't think `@Nullable` would make sense here as people could user the other method for particles that don't require options.